airbyte_cdk/sources/declarative/transformations/dpath_flatten_fields.py

```diff
@@ -24,17 +32,43 @@ class DpathFlattenFields(RecordTransformation):
     parameters: InitVar[Mapping[str, Any]]
     delete_origin_value: bool = False
     replace_record: bool = False
+    key_transformation: Union[KeyTransformation, None] = None
 
     def __post_init__(self, parameters: Mapping[str, Any]) -> None:
+        self._parameters = parameters
         self._field_path = [
-            InterpolatedString.create(path, parameters=parameters) for path in self.field_path
+            InterpolatedString.create(path, parameters=self._parameters) for path in self.field_path
         ]
         for path_index in range(len(self.field_path)):
             if isinstance(self.field_path[path_index], str):
                 self._field_path[path_index] = InterpolatedString.create(
-                    self.field_path[path_index], parameters=parameters
+                    self.field_path[path_index], parameters=self._parameters
                 )
 
+    def _apply_key_transformation(self, extracted: Mapping[str, Any]) -> Mapping[str, Any]:
+        if self.key_transformation:
+            if self.key_transformation.prefix:
+                prefix = InterpolatedString.create(
+                    self.key_transformation.prefix, parameters=self._parameters
+                ).eval(config=self.config)
+                extracted = {f"{prefix}{key}": value for key, value in extracted.items()}
+
+            if self.key_transformation.suffix:
+                suffix = InterpolatedString.create(
```

Comment:
Instead of instantiating the `InterpolatedString` for both the prefix and suffix every time we call `transform()` what I think we should do is use a `__post_init__()` method on the `KeyTransformation` class that takes in the prefix and suffix strings and instantiates InterpolatedStrings. And then from this method we should just need to call `self.key_transformation.prefix.eval()`.

Can we also add a check so that if `prefix.eval()` (+suffix) is an empty string or None, then we should not perform the transformation

Reply:
added `__post_init__()` and updated checks in `_apply_key_transformation()`